src/Turtle/Prelude.hs

```diff
@@ -1090,6 +1093,32 @@ time io = do
 hostname :: MonadIO io => io Text
 hostname = liftIO (fmap Text.pack getHostName)
 
+-- | Show the full path of an executable file
+which :: MonadIO io => FilePath -> io (Maybe FilePath)
+which cmd = fold go Control.Foldl.head
```

Comment:
I was wondering if it would be worth just returning `go` (i.e. change the result to `Shell FilePath`).  This would allow the user to retrieve all matching programs on the path.  For the common case they could wrap `which` in `fold which Control.Foldl.head`, but if they wanted even better error handling they could do something like:

```haskell
example = do
    paths <- fold go Control.Foldl.list
    case paths of
        []     -> ... -- Warn the user that no matches are found
        [path] -> ... -- Do something with the path
        _      -> ... -- Warn the user that there are multiple matches
```

I don't feel strongly about this, though